src/modules/fundingManager/depositVault/FM_DepositVault_v1.sol

```diff
@@ -0,0 +1,130 @@
+import {IFM_DepositVault_v1} from
+    "@fm/depositVault/interfaces/IFM_DepositVault_v1.sol";
+
+// Internal Dependencies
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+
+import {Initializable} from "@oz-up/proxy/utils/Initializable.sol";
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/extensions/IERC20Metadata.sol";
+
+// External Dependencies
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   Deposit Vault Funding Manager
+ *
+ * @notice  This contract allows users to deposit tokens to fund the workflow.
+ *
+ * @dev     Implements {IFundingManager_v1} interface.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+contract FM_DepositVault_v1 is
+    IFundingManager_v1,
+    IFM_DepositVault_v1,
+    Module_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId == type(IFundingManager_v1).interfaceId
+            || interfaceId == type(IFM_DepositVault_v1).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+
+    /// @dev The token that is deposited.
+    IERC20 private _token;
+
+    /// @dev Storage gap for future upgrades
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Init Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata,
+        bytes memory configData
+    ) external override initializer {
+        __Module_init(orchestrator_, metadata);
+
+        address orchestratorTokenAddress = abi.decode(configData, (address));
+        _token = IERC20(orchestratorTokenAddress);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public View Functions
+
+    /// @inheritdoc IFundingManager_v1
+    function token() public view returns (IERC20) {
+        return _token;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function deposit(uint amount) external {
+        address from = _msgSender();
+        _deposit(from, amount);
+    }
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function depositFor(address from, uint amount) external {
+        _deposit(from, amount);
```

Comment:
Missing a check that the from address is not address(this)

Reply:
Also discussed above ([here](https://github.com/InverterNetwork/contracts/pull/626#discussion_r1707911078)). I'd argue that this would just be a waste of gas to check, as a self-deposit can only happen if the contract has for some reason self-approved before, and even then it's just a weird `Deposit` being emitted with a false value, no funds are moved.

Reply:
I mostly commented this because it appeared as a requirement in the documentation, but I don't think it's so necessary

Reply:
Okay, makes sense, thank you 🙏 I removed the comment itself I've removed it in [this commit](https://github.com/InverterNetwork/contracts/pull/646/commits/a31b5938349b9456298d07d30f7f8b900179ea23), so we won't need/add a check itself here. The PR handling all fixes is #626.